webidl/Generate.hx

```diff
@@ -412,26 +412,32 @@ template<typename T> pref<T> *_alloc_const( const T *value ) {
 		var outFiles = [];
 		sources.push(lib+".cpp");
 		for( cfile in sources ) {
-			var out = cfile.substr(0, -4) + ".bc";
-			var args = params.concat(["-c", cfile, "-o", out]);
+			var sourcePath = if(sys.FileSystem.exists(opts.out+ "/" + cfile)){
+				opts.out + "/" + cfile;
+			} else { cfile; }
+
+			var out = opts.out + "/" + cfile.substr(0, -4) + ".bc";
+			var args = params.concat(["-c", sourcePath, "-o", out, "-I" + Sys.getCwd()]);
 			command( emcc, args);
 			outFiles.push(out);
 		}
 
 		// link : because too many files, generate Makefile
 		var tmp = "Makefile.tmp";
+		var libPath = opts.out + "/" + lib;
+
 		var args = params.concat([
-			"-s", 'EXPORT_NAME="\'$lib\'"', "-s", "MODULARIZE=1",
+			//"-s", 'EXPORT_NAME="\'$lib\'"', "-s", "MODULARIZE=1", "-O0", "-s", "USE_WEBGL2=1", "-s", "FULL_ES3=1", "-s", "NO_EXIT_RUNTIME=1",
+			"-s", 'EXPORT_NAME="\'$lib\'"', "-s", "MODULARIZE=1", "-O0", "-s", "NO_EXIT_RUNTIME=1",
 			"--memory-init-file", "0",
-			"-o", '$lib.js'
+			"-o", '$libPath.js'
 		]);
 		var output = "SOURCES = " + outFiles.join(" ") + "\n";
 		output += "all:\n";
 		output += "\t"+emcc+" $(SOURCES) " + args.join(" ");
 		sys.io.File.saveContent(tmp, output);
 		command("make", ["-f", tmp]);
 		sys.FileSystem.deleteFile(tmp);
+		sys.io.File.copy("index.html", opts.out+"/index.html");

```

Comment:
I see that this explains the two `index.html` files. If we have the favicon already in the webRoot folder when it builds I don't see any reason that we can't have the `index.html` file in there as well and leave out this copy.

Reply:
Sure.. I don't mind either way..

Reply:
In order to preserve the sources separate from the output dir concept, which I think is good, we could create like a "bundled" directory maybe and add that to the `webidl.json` options?